document/FileFormat.mdk

```diff
@@ -2408,23 +2412,132 @@ This allows tools to include information about the stack frame that is not expli
+with one exception, described in &sect;[#annotatedCodeLocation-logicalLocationKey].
+
+### `logicalLocationKey` property { #annotatedCodeLocation-logicalLocationKey }
+
+An `annotatedCodeLocation` object may contain a property named `logicalLocationKey` whose value is a string.
+If present, this string shall be equal to the name of one of the properties on the `run.logicalLocations` object (&sect;[#run-logicalLocations]),
+which provides additional information about the logical location specified by `fullyQualifiedLogicalName` (&sect;[#stackFrame-fullyQualifiedLogicalName]).
+For more information about the purpose of this property, see &sect;[#location-fullyQualifiedLogicalName].
+
+### `module` property { #annotatedCodeLocation-module }
+
+An `annotatedCodeLocation` object may contain a property named `module` whose value is a string
+containing the name of the module that contains the code location specified by this object.
+
+### `threadId` property { #annotatedCodeLocation-threadId }
+
+An `annotatedCodeLocation` object may contain a property named `threadId` whose value is an integer which
+identifies that thread that was executing when the execution of a code flow reached the location specified by this object.
+If this `annotatedCodeLocation` does not occur within a `codeFlow`, the `threadId` property shall be absent.
+
 ### `message` property { #annotatedCodeLocation-message }
 
 An `annotatedCodeLocation` object may contain a property named `message` whose value is a string that describes
 the significance of this location within a particular context.
 
+### `kind` property { #annotatedCodeLocation-kind }
+
+An `annotatedCodeLocation` object may contain a property named `kind` whose value is a string that provides
+a descriptive identifier that categorizes the annotation.
+
+If present, the `kind` property shall have one of the following values, with the specified meanings:
+
+* `"alias"`: This location defines an additional name for a variable defined in a declaration.
+
+* `"branch": At this location, a branch in the execution path occurred.
+
+* `"call"`: This location is the site of a function or method call.
+
+* `"callReturn"`: This location is the target of a return from a function or method.
+
+* `"declaration"`: The location declares a variable.
+
+* `"functionEnter"`: This location is an entry point to a function or method.
+
+* `"functionExit": This location is an exit point from a function or method (for example, a return statement).
+
+* `"sanitizer"`: At this location,  data that entered the system from outside (for example, from user input)
+is ensured to be safe.
+
+* `"sink"`: At this location, data leaves the system (for example, by being written to the console or sent over a network).
+
+* `"source"`: At this location, data enters the system (for example, by being provided by a user or read from a file on disk).
```

Comment:
your general definitions are correct. in practice, we tend to track sources of _untrusted_ data. it would be good to keep your general definitions but allude to the practical uses in examples, since they tend to be the focus of analysis tools (as opposed to tooling that tracks general flow of data)